src/config/config.h

```diff
@@ -190,19 +190,46 @@ std::shared_ptr<Config<data_t>> ConfigFile::Get(const std::string& config_name,
     return config_ptr;
 }
 
+#define __EXTERN_CONFIG_TYPE(type)                                                                   \
```

Comment:
You can use a macro in ifdef XXX to control the existence of extern, and define XXX in cpp before including this header.

See `_CRIS_MATH_INST_SAR_TMPL` in cris

Reply:
Abusing macros is bad, leaking these types of macros externally to the cpp files is even worse.

Reply:
There won't be any leak if you undef, though I actually prefer leaking because undef can lead of silent collision.